### PR TITLE
bt-512: Add ability to view password during sign in up

### LINKED ISSUE #512: Add ability to view password during sign in/up
Add ability (eye icon) to view password in sign in/up to improve UX.

### PR DESCRIPTION
**Pull request by illia-dobro**
visible
novisib

### CODE REVIEW

File: mobile/src/bundles/auth/components/sign-in-form/sign-in-form.tsx [hunk 1]
```diff
@@ -11,9 +11,14 @@ import {
...
 import { AuthScreenName, TextCategory } from '~/bundles/common/enums/enums';
-import { useAppForm, useCallback } from '~/bundles/common/hooks/hooks';
+import {
+    useAppForm,
+    useCallback,
+    useState,
+} from '~/bundles/common/hooks/hooks';
 import { globalStyles } from '~/bundles/common/styles/styles';
 
+import { PasswordVisibilityToggle } from '../components';
```
Comment (mskmee) [inline mobile/src/bundles/auth/components/sign-in-form/sign-in-form.tsx]: it's better to use absolute path here

File: mobile/src/bundles/auth/components/sign-up-form/sign-up-form.tsx [hunk 2]
```diff
@@ -17,9 +17,15 @@ import {
...
-import { useAppForm, useCallback, useMemo } from '~/bundles/common/hooks/hooks';
+import {
+    useAppForm,
+    useCallback,
+    useMemo,
+    useState,
+} from '~/bundles/common/hooks/hooks';
 import { globalStyles } from '~/bundles/common/styles/styles';
 
+import { PasswordVisibilityToggle } from '../components';
```
Comment (mskmee) [inline mobile/src/bundles/auth/components/sign-up-form/sign-up-form.tsx]: same here

**Review by nikita-remeslov — APPROVED** [review]

File: mobile/src/bundles/auth/components/password-visibility-toggle/password-visibility-toggle.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import Icon from 'react-native-vector-icons/MaterialIcons';
```
Comment (katerynakostikova) [inline mobile/src/bundles/auth/components/password-visibility-toggle/password-visibility-toggle.tsx]: MaterialIcon component

File: mobile/src/bundles/auth/components/password-visibility-toggle/password-visibility-toggle.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import { TouchableOpacity } from '~/bundles/common/components/components';
+import { Color, IconName } from '~/bundles/common/enums/enums';
+
+import { styles } from './styles';
+
+type Properties = {
+    isPasswordVisible: boolean;
+    setIsPasswordVisible: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (katerynakostikova) [inline mobile/src/bundles/auth/components/password-visibility-toggle/password-visibility-toggle.tsx]: onChangeVisibility: () => void

File: mobile/src/bundles/auth/components/sign-in-form/styles.ts [hunk 5]
```diff
@@ -28,6 +28,11 @@ const styles = StyleSheet.create({
     formWrapper: {
         gap: 25,
     },
+    icon: {
+        position: 'absolute',
+        right: 20,
+        top: 48,
+    },
```
Comment (katerynakostikova) [inline mobile/src/bundles/auth/components/sign-in-form/styles.ts]: 2 repeatable styles?

File: mobile/src/bundles/auth/components/sign-in-form/sign-in-form.tsx [hunk 6]
```diff
@@ -29,6 +34,9 @@ const SignInForm: React.FC<Properties> = ({ onSubmit }) => {
     const handleFormSubmit = useCallback((): void => {
         void handleSubmit(onSubmit)();
     }, [handleSubmit, onSubmit]);
+
+    const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+
```
Comment (likeri29) [inline mobile/src/bundles/auth/components/sign-in-form/sign-in-form.tsx:39]: Maube it's better to use our useVisibility hook?

**Comment by katerynakostikova** [comment]
Fix other students comments and I'll approve

**Review by katerynakostikova — APPROVED** [review]

**Pull request merged**